src/search/indexing/mod.rs

```diff
@@ -0,0 +1,133 @@
+/// This module is used for the indexing from any source.
+pub mod curseforge_import;
+pub mod local_import;
+
+use thiserror::Error;
+use crate::search::models::SearchMod;
+use meilisearch_sdk::client::Client;
+use std::collections::{HashMap, VecDeque};
+use meilisearch_sdk::settings::Settings;
+use crate::search::indexing::curseforge_import::index_curseforge;
+use crate::search::indexing::local_import::index_local;
+use crate::database::DatabaseError;
+
+#[derive(Error, Debug)]
+pub enum IndexingError {
+    #[error("Error while connection to the meilisearch database")]
+    IndexDBError(),
+    #[error("Error while connecting to the local server")]
+    LocalDatabaseError(#[from] mongodb::error::Error),
+    #[error("Error while accessing the data from remote")]
+    RemoteWebsiteError(#[from] reqwest::Error),
+    #[error("Error while serializing or deserializing json")]
+    SerDeError(#[from] serde_json::Error),
+    #[error("Error while parsing float")]
+    FloatParsingError(#[from] std::num::ParseFloatError),
+    #[error("Error while parsing float")]
+    IntParsingError(#[from] std::num::ParseIntError),
+    #[error("Error while parsing GSON")]
+    DatabaseError(#[from] DatabaseError)
+}
+
+pub async fn index_mods(db: mongodb::Client) -> Result<(), IndexingError>{
+    // Check if the index exists
+    let client = Client::new("http://localhost:7700", "");
```

Comment:
This should probably use `dotenv` with a default instead of hardcoding the url

Reply:
will do